modules/tensor_mechanics/src/auxkernels/RankTwoScalarAux.C

```diff
@@ -47,9 +47,46 @@ RankTwoScalarAux::computeValue()
    case 3:
      val = _tensor[_qp].L2norm();
      break;
+   case 4:
+   case 5:
+   case 6:
+     val = calcEigenValues();
+     break;
    default:
-     mooseError("RankTwoScalarAux Error: Pass valid scalar type - VonMisesStress, EquivalentPlasticStrain, Hydrostatic, L2norm");
+     mooseError("RankTwoScalarAux Error: Pass valid scalar type - VonMisesStress, EquivalentPlasticStrain, Hydrostatic, L2norm MaxPrincipal MidPrincipal MinPrincipal");
   }
   return val;
 }
 
+Real
+RankTwoScalarAux::calcEigenValues()
+{
+  std::vector<Real> eigval(LIBMESH_DIM);
+  Real val = 0.0;
+  unsigned int max_index = 2;
+  unsigned int mid_index = 1;
+  unsigned int min_index = 0;
+
+  if (LIBMESH_DIM == 2)
```

Comment:
Uhm, should we maybe catch the super-crazy case where someone compiles MOOSE with LIBMESH_DIM = 4?!